packages/extensionmanager/src/widget.tsx

```diff
@@ -318,7 +326,11 @@ namespace ListView {
      *
      * Not provided if actions are not allowed.
      */
-    performAction?: (action: Action, entry: IEntry) => void;
+    performAction?: (
+      action: Action,
+      entry: IEntry,
+      actionOptions: IActionOptions
```

Comment:
let's make this one optional:

```suggestion
      actionOptions?: IActionOptions
```